src/scripts/main.js

```diff
@@ -1,3 +1,76 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field tbody');
+const container = document.querySelector('.container');
+const [
+  appendRow,
+  removeRow,
+  appendColumn,
+  removeColumn,
+] = document.querySelectorAll('.button');
+
+container.addEventListener('click', {
+  handleEvent(ev) {
+    if (!ev.target.classList.contains('button')) {
+      return;
+    }
+
+    const buttonAction = ev.target.classList[0];
+    const method = capitalized(buttonAction);
+
+    this[method]();
+
+    const rowsCount = table.children.length;
+    const columnsCount = table.lastElementChild.children.length;
+    const maxCells = 10;
+    const minCells = 2;
+
+    appendRow.disabled = (rowsCount >= maxCells);
+    removeRow.disabled = (rowsCount <= minCells);
+    appendColumn.disabled = (columnsCount >= maxCells);
+    removeColumn.disabled = (columnsCount <= minCells);
+  },
+
+  appendRow() {
```

Comment:
Do you really need to store all the methods inside the event listener?
Move them out of its scope and pass only handle event function to the event listener